src/platform/react-hooks/src/ChannelProvider.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useLayoutEffect, useEffect } from 'react';
+import * as Ably from 'ably';
+import { getContext } from './AblyProvider.js';
+import * as Utils from './utils/utils.js';
+import { channelOptionsWithAgent } from './AblyReactHooks.js';
+
+interface ChannelProviderProps {
+  id?: string;
+  channelName: string;
+  options?: Ably.Types.ChannelOptions;
+  deriveOptions?: Ably.Types.DeriveOptions;
+  children?: React.ReactNode | React.ReactNode[] | null;
+}
+
+export const ChannelProvider = ({
+  id = 'default',
```

Comment:
Would it make sense to just use the channel name as the ID here? Maybe derived channels would make this more complicated but otherwise, channel names should be unique per-channel so we might not need a separate id?

Reply:
oh, looks like it's not clear, it's not a provider ID, it's ably instance ID (inherited from AblyProvider)

Reply:
Ah, my mistake 👍